### ISSUE TITLE
Disgaea 4 Complete+ (1233880)

### ISSUE THREAD

**Issue by chriztheanvill** [ValveSoftware/Proton#5164]
# Compatibility Report
- Name of the game with compatibility issues: Disgaea 4 Complete+. Linux
- Steam AppID of the game: 1233880

## System Information
- GPU: Intel Corporation UHD Graphics 620 (rev 07)
- Proton version: 6.3-6 and Experimental

Operating System Version:
    Ubuntu 21.04 (64 bit)
    Kernel Name:  Linux
    Kernel Version:  5.11.0-35-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12011000
    X Window Manager:  KWin
    Steam Runtime Version:  steam-runtime_0.20210906.1
Video Card:
    Driver:  Intel Mesa Intel(R) UHD Graphics 620 (KBL GT2)
    Driver Version:  4.6 (Compatibility Profile) Mesa 21.0.3
    OpenGL Version: 4.6

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 
After the new update of Experimental, can not pass the cinema, even if I push many times: k or l. This was the trick to play on recent releases.
Now I can not play on 6.3-6 or Experimental.

## Reproduction
I think is the new update of Experimental.
[steam-1233880.log](https://github.com/ValveSoftware/Proton/files/7189694/steam-1233880.log)

**Comment by kisak-valve** [comment]
Hello @chriztheanvill, what's the most recent Proton version that you know you are able to play the game with?

**Comment by chriztheanvill** [comment]
Hi kisak. With Proton: 6.3-6, and with the previous Experimental version.
Its playable 5.13-6, but has no sound, and it needs to do the trick of: pressing many times the key: k or l to skip the first cinema.

Edit: I made a post on reddit about this, and there is another user that has the same problem.
https://www.reddit.com/r/Disgaea/comments/pqkdli/disgaea_4_complete_proton_linux_breaks/

**Comment by chriztheanvill** [comment]
Hi: I've just followed this steps and works, you need to use proton-ge and protontricks. I do not know if would helps:

- Delete the compatdata folder of your game ~/compatdata/1233880/ if you already had launched the game.
    - In my case it was in: ~/.steam/debian-installation/steamapps/compatdata/1233880
- Install Proton-GE, specifically, Proton-5.9-GE-6-ST. Couldn't get the latest ones to work.
- Launch the game with the specific Proton-GE version then close it.
- Install protontricks and launch the command in the console.
-  protontricks 1233880 winecfg
- Go to the libraries tab and add the following then close it.
    - d3dx9_43
    - mfplat
    - wmp
   - xaudio2_9

- Add the following launch options ( on steam: Properties > General > Launch Options )

 PROTON_FORCE_LARGE_ADDRESS_AWARE=1 PROTON_NO_ESYNC=1 PULSE_LATENCY_MSEC=60 %command%

- Launch the game and spam 'k' and the game should boot up with sound!

**Comment by chriztheanvill** [comment]
Now with the experimental, and enabling the Shaders Pre-Caching: Settings > Shaders Pre-Caching > Allow background processing of Vulkan shaders.
Works, no intro video, but works with audio.

**Comment by ipr** [comment]
The button-configuration for gamepad seems to be Nintendo-style (A and B buttons reversed, confirm and cancel, while using PS4 controller) and the text during intro is skipped for some reason as if a button was stuck. So this means you don't have time to read the intro text.

**Comment by chriztheanvill** [comment]
With the current experimental, Everything works Excellent, even the intro.

**Comment by ipr** [comment]
> With the current experimental, Everything works Excellent, even the intro.

Nope, audio does not work. (referring to media related issue: #1464)

**Comment by chriztheanvill** [comment]
> > With the current experimental, Everything works Excellent, even the intro.
> 
> Nope, audio does not work. (referring to media related issue: #1464)

I've just play it now and everything works. Even, works better han D5.

**Comment by hasezoey** [comment]
i have tested the game yesterday, where it didnt work, now i tested it again (where the only change was the steam update `1644012199` (the update that cleans shader cache on driver update)), and the intro cutscene works, but everything after this "crashes" plus other problems:
- on each start of the game (even when compatdata already exists), steam tries to do the first-time-setup step `Installing: Microsoft DirectX for Windows`
- resizing the game makes the game display a blackscreen
- the intro cutscene works now, but after that it is still the 100% CPU issue and does not progress to the main menu (skipping does also not work and brings it directly to the 100% CPU issue) (Proton 6.3-8)
- when having viewed the intro once, on each start afterwards it goes directly to the 100% CPU issue (Proton 6.3-8)
- once the 100% CPU issue is in effect (or not progressing with a blackscreen / infinite loadingscreen), closing the window with SIGTERM / X in the corner / ALT+F4 does not work and i need to use SIGKILL
- the loading screen that should take me to the menu is infinite and does nothing (if it even shows, because sometimes it does not) (also cannot be closed without SIGKILL) (Proton Experimental)
- on Proton Experimental, i could not encounter the 100% CPU issue, but it still does not progress

TL;DR: intro cutscene works now, but still cannot access the main menu "out-of-the-box"

other information:
- Manjaro Linux 21.2.2 KDE Plasma Wayland + Pipewire
- Proton 6.3-8 1638789187 proton-6.3-8c
- Proton Experimental 1642614452 experimental-6.3-20220119
- Kernel: 5.16.2-1-MANJARO
- CPU: Intel i7-7700k
- GPU: AMD Vega 64 (VEGA10, DRM 3.44.0, 5.16.2-1-MANJARO, LLVM 13.0.0)

[Proton Experimental infinite loading screen log](https://github.com/ValveSoftware/Proton/files/8007694/steam-1233880.log)
[Proton Experimental Skip intro scene, loading screen not showing log](https://github.com/ValveSoftware/Proton/files/8007698/steam-1233880.log)

**Comment by chriztheanvill** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/5164#issuecomment-1030587007

Try on Steam program
Steam > Settings > Shader Pre-Caching > 
And enable both checkbox.

**Comment by hasezoey** [comment]
> Try on Steam program Steam > Settings > Shader Pre-Caching > And enable both checkbox.

sorry that i didnt mention it before, but both of them are already enabled, and also my steps before re-trying a (complete fresh) start of the game are:
- stop steam
- cleanup the compat data
- cleanup the shader cache
- start steam
- wait for shader cache to be downloaded
- (sometimes) verify game file integrity

if you replied because of `(the update that cleans shader cache on driver update)`, then that was only to mention what update (aside from the number) i meant

**Comment by hasezoey** [comment]
Update to my previous statement:
it seems like it was another case of problems with sshfs, once i had the game moved to a local drive, it worked without problems (no infinite loading screen), though some weird issues still remain:
- when trying to exit the game (with the X button of a window), while in a loading screen (especially when using sshfs), it does not exit and needs to be killed (goes to 100% CPU after having pushed exit, my guess it is waiting for something IO related)
- window size that is select-able is limited to 1920x1080, but manually resizing the window works
- the game has a very weird keyboard layout (i didnt use a controller for quick testing)

keyboard layout for anyone that is confused:
- Q/E turn pages back and forth
- K apply changed
- L go back
- A/D to change values
- ESC / ENTER / SPACE / ARROW KEYS did nothing

**Comment by chriztheanvill** [comment]
With the experimental, disgaea 4, 5 cant pass the initial video, it closes the game.

**Comment by ipr** [comment]
Main menu does not load with Proton 7.0 and experimental: game is infinitely waiting for something after intro video.

Edit: not using sshfs

[steam-1233880.log.gz](https://github.com/ValveSoftware/Proton/files/9459854/steam-1233880.log.gz)

Edit, update: after running update (there was Vulkan update waiting) now loads into main menu with Proton experimental. Also audio is playing back correctly (there was problem before with earlier Proton).

**Comment by hasezoey** [comment]
just tried again on a updated system, but this time it does not even launch. (not showing any window after "Starting Game" from steam)

tested: Proton Experimental (`1670512927 experimental-7.0-20221208`), Proton 7.0-5, Proton 6.3-8 (`1655928813 proton-6.3-8f`)
Edit: also tested Proton 7.0-6 RC (`1671220184 proton-7.0-6-rc5`)

[Proton Experimental steam-1233880.log](https://github.com/ValveSoftware/Proton/files/10254033/steam-1233880.log)

i have already tried removing the compatdata for the appid and also verified all files from the game, proton and runtime

<details>
<summary>Steam System Information</summary>

```txt
Computer Information:
    Manufacturer:  ASUSTeK COMPUTER INC.
    Model:  ROG STRIX B550-F GAMING
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  AuthenticAMD
    CPU Brand:  AMD Ryzen 7 5800X 8-Core Processor             
    CPU Family:  0x19
    CPU Model:  0x21
    CPU Stepping:  0x2
    CPU Type:  0x0
    Speed:  4850 Mhz
    16 logical processors
    8 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Supported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    AVX2:  Supported
    AVX512F:  Unsupported
    AVX512PF:  Unsupported
    AVX512ER:  Unsupported
    AVX512CD:  Unsupported
    AVX512VNNI:  Unsupported
    SHA:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    "Manjaro Linux" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  6.0.11-1-MANJARO
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12201005
    X Window Manager:  KWin
    Steam Runtime Version:  steam-runtime_0.20221019.0

Video Card:
    Driver:  AMD AMD Radeon RX Vega (vega10, LLVM 14.0.6, DRM 3.48, 6.0.11-1-MANJARO)
    Driver Version:  4.6 (Compatibility Profile) Mesa 22.2.4
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 74 Hz
    VendorID:  0x1002
    DeviceID:  0x687f
    Revision Not Detected
    Number of Monitors:  3
    Number of Logical Video Cards:  1
    Primary Display Resolution:  2560 x 1440
    Desktop Resolution: 6400 x 1440
    Primary Display Size: 27.95" x 15.75" (32.05" diag)
                                            71.0cm x 40.0cm (81.4cm diag)
    Primary VRAM: 8192 MB

Sound card:
    Audio device: ATI R6xx HDMI

Memory:
    RAM:  15906 MB

VR Hardware:
    VR Headset: None detected

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Total Hard Disk Space Available:  921069 MB
    Largest Free Hard Disk Block:  837877 MB

Storage:
    Number of SSDs: 7
    SSD sizes: 1000G,1000G,1000G,500G,320G,240G,0B
    Number of HDDs: 0
```

</details>

**Comment by hasezoey** [comment]
today it would launch, so it was probably related to "steam restarting fixes proton issues"

though the issue about closing / exiting not actually exiting are still present
the same log also shows that "infinite loading screen" after the intro video (which played fine)
[log of not exiting steam-1233880.log](https://github.com/ValveSoftware/Proton/files/10260132/steam-1233880.log)
Note: the log shows usage of proton 7.0-5, but also happens with Experimental (`1670512927 experimental-7.0-20221208`)
Note: one CPU core is pinned to 100% after trying to exit via the X of the window or ingame (if i could actually get there)
Note: i am not using sshfs this time, it is direct btrfs on a ssd (SATA)
Note: also tried deleting the compdata again, nothing changed

btw, resizing the window while the intro is playing only shows a black screen but the audio of the intro still plays

and i cannot get to the main-menu

---

also i have noticed that on wayland it creates 2 windows: one with the game logo which is actually the game (preview from kde hover) and one with the fallback xwayland logo with no output and both windows do not have a title
![1](https://user-images.githubusercontent.com/10911626/208453402-8ac5f0b1-5b3a-4569-aeb1-bfc74c3effe1.png)
![2](https://user-images.githubusercontent.com/10911626/208453393-4745c5b2-750c-4231-a781-9857d82be38e.png)


---

Update:
while doing the tests above, i had a controller plugged in (which also worked with skipping the logos and intro) (`XBox One Wireless (Wired) Controller`), now i have tried to launch with a unplugged controller and it did not load infinitely anymore and once in the main-menu i tried re-plugging the controller which just worked and the game also picked it up and let me use it
Note: Controller is currently set via steam-input not directly

Another Update:
When switching to "Full Screen - Borderless", the window just keeps flickering, while the same messages in the log are printed
[log of window flickering in borderless steam-1233880.log](https://github.com/ValveSoftware/Proton/files/10260287/steam-1233880.log)

Another Note:
when trying to force the game to fullscreen (via KDE's kwin fullscreen shortcut), the game just reverts to its previous window state

**Comment by chriztheanvill** [comment]
Today I tried play Disgaea 4 and sometimes can be played or just stay in loading.
Just switched the linux distro and PC:

Debian SID + Cinnamon.
Shell: zsh 5.9
DE: GNOME 43.1
WM: Muffin
CPU: AMD Ryzen 7 5700G with Radeon Graphics @ 16x 4.673GHz
GPU: AMD Radeon Graphics (renoir, LLVM 15.0.6, DRM 3.48, 6.0.0-6-amd64)

**Comment by chriztheanvill** [comment]
Hi.
Just bought the steam deck and same thing.
Is there a way to fix this ?
I am a C++ dev, I could help in this game.

**Comment by ipr** [comment]
> Hi. Just bought the steam deck and same thing. Is there a way to fix this ? I am a C++ dev, I could help in this game.

It works for me on the Deck. Have you check for updates? (under system settings in Steam)

First loading often takes longer as shaders are built then.

Note that while in-game, B/A buttons are randomly swapped for some reason (confirm is on B most of the time)

**Comment by chriztheanvill** [comment]
> > Hi. Just bought the steam deck and same thing. Is there a way to fix this ? I am a C++ dev, I could help in this game.
> 
> It works for me on the Deck. Have you check for updates? (under system settings in Steam)
> 
> First loading often takes longer as shaders are built then.
> 
> Note that while in-game, B/A buttons are randomly swapped for some reason (confirm is on B most of the time)

Sometimes start the game, and sometimes dont.

**Comment by chriztheanvill** [comment]
_WOW, LOOKS LIKE THE LATEST PROTON WORKS EXCELLENT WITH DISGAEA!!!!!!!!!!!!!!!!!!!!!!!!!_

Edit:
Na, still the same.

**Comment by ipr** [comment]
With Proton 8.0-4 on Deck game crashes when entering world from dimension guide
[steam-1233880.log.gz](https://github.com/ValveSoftware/Proton/files/12887652/steam-1233880.log.gz)

**Comment by ipr** [comment]
I'll elaborate a bit more.
Entering one of the early missions (tutorials) from the dimension guide has regressed so that is crashes the game. Proton 7.0 did not crash, but 8.0 does not crash every time either.
So it is a regression but repeating crash might not happen every time.

**Comment by kisak-valve** [comment]
Hello @ipr, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by ipr** [comment]
> Hello @ipr, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

It is on Steam Deck, log is in message above (hard to type on-screen so I'm typing this on different computer).

OS build 20230915.100

**Comment by ipr** [comment]
You do need to start a new game so that the tutorial levels are opened. Normal levels don't have this issue.
The tutorial levels have something different in them.

**Comment by ipr** [comment]
Hmm.. Proton experimental started crashin during opening animation. That used to work better in-game too
[steam-1233880.log.gz](https://github.com/ValveSoftware/Proton/files/12910999/steam-1233880.log.gz)
this is on Deck with 3.5.1 OS

Interesting bit at the end of the log:
``
5144.731:0134:0208:err:seh:call_stack_handlers invalid frame 000000007544E648 (0000000075252000-0000000075350000)
5144.731:0134:0208:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
wine client error:208: write: Bad file descriptor
``

**Comment by ipr** [comment]
Works fine on SteamOS 3.6 preview with Proton 9.0-1.

Much smoother animations and better response too. Nice.

**Comment by chriztheanvill** [comment]
> Works fine on SteamOS 3.6 preview with Proton 9.0-1.
> 
> Much smoother animations and better response too. Nice.

Do you need this commands to pass the first scenema ?
`PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1 %command%`

**Comment by ipr** [comment]
> > Works fine on SteamOS 3.6 preview with Proton 9.0-1.
> > Much smoother animations and better response too. Nice.
> 
> Do you need this commands to pass the first scenema ? `PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1 %command%`

No.

**Comment by TheDukeofErl** [comment]
Is there an actual workaround for the infinite "Now Loading" screen other than simply relaunching the game over and over until it gets past that screen? This happens on the Steam Deck, regardless of Proton version used, as well as on my normal PC, regardless of Proton version. It seems that this is an abundantly common issue, judging by posts on the Steam forums, the game's reviews, protondb, and here.

This seems to have been an issue for a long time, including when the game was "Verified" on Steam. At a minimum, is there a way to remove the game's "Verified" status? It's somewhat confusing to me that the game is marked that way at all when it simply doesn't function well over half the time.

**Comment by chriztheanvill** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/5164#issuecomment-2106358454

In Launch Options `PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1 %command%`

**Comment by TheDukeofErl** [comment]
> In Launch Options `PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1 %command%`

Phew, that does it. Using both Steam Deck with whatever version of Proton Valve recommends as well as on Arch Linux with Proton 9. Thanks!

**Comment by waspennator** [comment]
> > Replying to [#5164 (comment)](https://github.com/ValveSoftware/Proton/issues/5164#issuecomment-2106358454)
> 
> In Launch Options `PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1 %command%`

Feel like these launch options should be included in proton by default for this game. I don't have the game on my end, but I still see many reports saying it needs these commands to consistently work.
